### ISSUE TITLE
Products not getting indexed when using SMILE as search engine

### ISSUE THREAD

**Issue by ejaz-alam** [Smile-SA/elasticsuite#2912]
I'm currently experiencing an issue where my products are not getting indexed when using SMILE as the search engine, while categories are getting indexed without any issues. When I switch to using Elasticsearch as the search engine, the products are getting indexed correctly.

Steps to reproduce:

Enable SMILE as the search engine in the Magento 2 backend.
Add a new product in Magento 2.
Wait for the search indexer to run.
Search for the new product using the search bar on the frontend of the website.
Notice that the product does not appear in the search results.
Expected behavior: The new product should appear in the search results when using SMILE as the search engine.

Actual behavior: The new product does not appear in the search results when using SMILE as the search engine, but does appear when using Elasticsearch as the search engine.

Environment:

Magento version: 2.4.5
SMILE ElasticSuite version: 2.10.13
Elasticsearch version : 7.17
Additional information:

I have tried reindexing the search index multiple times, but the product still does not appear in the search results.
Categories are getting indexed correctly when using SMILE as the search engine.
When using Elasticsearch as the search engine, the products are getting indexed correctly.
Any help in resolving this issue would be greatly appreciated. Thanks in advance!

indices without smile as a search engine:
![image](https://user-images.githubusercontent.com/22393364/232384410-e889c5fd-70e7-4ab1-b2aa-667efa079fa2.png)

indices with smile as a search engine:
![image](https://user-images.githubusercontent.com/22393364/232385198-7ac3d227-5e33-421a-b407-43eb74f6056b.png)

**Comment by vahonc** [comment]
Hello @ejaz-alam,

Usually, everything should work as expected, probably there are configuration issues or conflicts with third-party modules.

Based on your screenshots, I assume that you are using an environment with multi websites/stores or you are using different M2 projects on the same server? 

What is the Indices Alias Name of your project?

![Screenshot from 2023-04-14 14-20-48](https://user-images.githubusercontent.com/67791361/232430807-70e936b5-141f-4b78-80e1-ffafa0810647.png)

BR,
Vadym

**Comment by ejaz-alam** [comment]
yes, we are using multistore/website. our project indices Alis is temp-m2.
![image](https://user-images.githubusercontent.com/22393364/232433392-6a66cc97-3129-4e7b-976b-43eda36a191b.png)

**Comment by vahonc** [comment]
@ejaz-alam, 

I noticed that all your *_catalog_product_* indices have size 226b which means that there is no data. Are you sure that you properly configure your products (visibility, relation between websites, etc)? Are they displayed correctly on the frontend if you go to some category?

Have you run a full reindex and check logs if there are some errors?

**Comment by ejaz-alam** [comment]
@vahonc , yes, I am sure because the database is the same as we were using. we just upgraded the Magento and also upgrade the smile module. after that, the products are not indexed. the problem is I am not seeing any errors. and yes I have run the full indexer.

**Comment by vahonc** [comment]
@ejaz-alam, 

What kind of migration did you do?

Based on the fact that all your `*_catalog_product_*` indexes are too small, it means that something is wrong with the product data. Perhaps this is due to the fall of the product attribute mapping. This is possible if something went wrong during the migration process or some kind of error occurred. So check the attributes of your products. Also, if you don't find any errors in the Magento logs, you can check the Elasticsearch logs in `/var/log/elasticsearch/`.

BR,
Vadym

**Comment by romainruaud** [comment]
You also have to check that all your categories are "is_anchor=1".

This is something that is enforced by Elasticsuite when creating new categories. But if you imported them, you should check that they are "is_anchor=1" otherwise they'll have no products attached.

Regards
